components/domains/steps/userInfoForm.tsx

```diff
@@ -0,0 +1,174 @@
+      case FormType.RENEW:
+        return "Renew Your domain(s)";
+      default:
+        return "";
+    }
+  };
+
+  function changeEmail(value: string): void {
+    setEmail(value);
+    setEmailError(isValidEmail(value) ? false : true);
+    updateFormState({ email: value });
+  }
+
+  function changeTokenId(value: number): void {
+    updateFormState({ tokenId: value });
+  }
+
+  function changeDuration(value: number): void {
+    if (isNaN(value) || value > maxYearsToRegister || value < 1) return;
+    updateFormState({
+      duration: value,
+      isUpselled: value === 1 ? true : false,
+    });
+  }
+
+  return (
+    <div className={styles.container}>
+      <div className={styles.card}>
+        <div className={styles.form}>
+          <div className="flex flex-col items-start gap-4 self-stretch">
+            <p className={styles.legend}>{type}</p>
+            <h3 className={styles.domain}>{getTitle()}</h3>
+          </div>
+          <div className="flex flex-col items-start gap-6 self-stretch">
+            {formState.needMetadata ? (
+              <TextField
+                helperText="Secure your domain's future and stay ahead with vital updates. Your email stays private with us, always."
+                label="Email address"
+                value={email}
+                onChange={(e) => changeEmail(e.target.value)}
+                color="secondary"
+                error={emailError}
+                errorMessage={"Please enter a valid email address"}
+                type="email"
+              />
+            ) : null}
+
+            {formState.needMetadata ? (
+              <SwissForm
+                isSwissResident={formState.isSwissResident}
+                onSwissResidentChange={() =>
+                  updateFormState({
+                    isSwissResident: !formState.isSwissResident,
+                  })
+                }
+              />
+            ) : null}
+            {type === FormType.REGISTER ? (
+              <SelectIdentity
+                tokenId={formState.tokenId}
+                changeTokenId={changeTokenId}
+              />
+            ) : null}
+            <NumberTextField
+              value={formState.duration}
+              label="Years to register (max 25 years)"
+              placeholder="years"
+              onChange={(e) => changeDuration(Number(e.target.value))}
+              incrementValue={() =>
+                changeDuration(
+                  formState.duration < maxYearsToRegister
+                    ? formState.duration + 1
+                    : formState.duration
+                )
+              }
+              decrementValue={() =>
+                changeDuration(
+                  formState.duration > 1
+                    ? formState.duration - 1
+                    : formState.duration
+                )
+              }
+              color="secondary"
+              required
+            />
```

Comment:
for both `incrementValue` & `decrementValue`could we add less inline logic and add this directly to the changeDuration function ?

you could add another argument to the `changeDuration` being the type (increment or decrement) for example